src/command_modules/azure-cli-component/azure/cli/command_modules/component/custom.py

```diff
@@ -132,7 +141,25 @@ def _install_or_update(package_list, link, private, pre):
     _run_pip(pip, pip_args)
 
 
-def update(private=False, pre=False, link=None, additional_components=None):
+def _verify_additional_components(components, private, allow_third_party):
+    # Don't verify as third party packages allowed or private server which we can't query
+    if allow_third_party or private:
+        return
+    third_party = []
+    first_party_component_names = [r['name']for r in _get_first_party_pypi_command_modules()]
+    for c in components:
+        if c not in first_party_component_names:
+            third_party.append(c)
+    if third_party:
+        raise CLIError("The following components are third party or not available"
```

Comment:
I suggest improve the error message meaning, `the following components ... are third party, which will not be installed unless --allow-third-party is specified`